build/scripts/GuardingV2ExtensionsHelper.psm1

```diff
@@ -251,4 +251,13 @@ function Test-IsStrictModeEnabled
     return $false
 }
 
+function GetCurrentBuildVersionFromMain() {
```

Comment:
It's not really the "build version" this function gets.

How about `GetMaxAllowedObsoleteVersion` plus some comments explaining why the main's major version is the max allowed obsolete version?